enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,562 @@
+in MCO to make changes without rebooting.
+
+Scaleout in this implementation (using static IPs on baremetal to demonstrate
+as many parts of the process as possible) would look something like this:
+
+* Add the new node's configuration to the machine-config for its role.
+  * `oc edit mc 10-br-ex-worker`
+* Create and apply a secret containing the minimal static IP configuration
+  needed for the node to pull ignition.
+  * `oc apply -f ./extraworker-secret.yaml`
+* Create and apply a BareMetalHost resource with the network secret attached
+  via the `preprovisioningNetworkDataName` field.
+```
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+spec:
+...
+  preprovisioningNetworkDataName: ostest-extraworker-0-network-config-secret
+```
+* Scale out the machineset to deploy the new node.
+  * `oc project openshift-machine-api`
+    `oc get machinesets`
+    `oc scale machineset [name of machineset] --replicas=[number of nodes desired after scaling]`
+
+The node will initially boot with the network configuration specified in
+`preprovisioningNetworkDataName` and then have the configuration in the
+machine-config applied.
+
+### API Extensions
+
+This will not directly modify the API of OpenShift. The API for this feature is
+implemented in Kubernetes-NMState, and this does not require any specific
+modifications in that project at this time.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+Because this gives the deployer a great deal of control over the bridge
+configuration for OVNK, they will have the ability to configure it incorrectly,
+which may cause problems either on initial deployment or at any point after.
+We will want to ensure that must-gather is collecting enough information about
+the bridge configuration for us to determine whether it is compliant with our
+guidelines, and we must have clear instructions on what is required for br-ex.
+
+Note, however, that if an incorrect bridge configuration is applied NMState
+will usually catch that and roll back the changes.
+
+### Drawbacks
+
+OpenShift has historically avoided per-host configuration for cluster nodes.
+This is a large departure from that philosophy, but that is unavoidable.
+Certain aspects of networking (notably static IPs) are by nature host-specific
+and cannot be handled any other way.
+
+Using a machine-config to do initial deployment and a
+NodeNetworkConfigurationPolicy to make day 2 changes will require a manual sync
+of the NMState configuration. Unfortunately this is unavoidable because the
+Kubernetes-NMState operator is not part of the core payload so it (and its
+CRDs) cannot be used for initial deployment.
+
```

Comment:
But we can change k-nmstate operator to read stuff under /etc/nmstate and convert it to NNCPs.

Reply:
That's an interesting idea. We could do that with the baremetal feature because only the nmconnection files made it to the host, but in this case we'll have the full NMState config available. It would be very interesting to add a feature to discover the existing config for the operator.